roles/kubernetes_setup/tasks/main.yml

```diff
@@ -33,6 +33,39 @@
     - "{{ kube_controller_manager }}"
     - "{{ kube_scheduler }}"
 
+- name: Define kube 1.7 master nodes images
+  when: ansible_distribution == "Fedora" and ansible_distribution_major_version > '26'
+  set_fact:
+    kube_maj_ver: "1.7"
+    kube_apiserver: "registry.fedoraproject.org/f27/kubernetes-apiserver"
+    kube_controller_manager: "registry.fedoraproject.org/f27/kubernetes-controller-manager"
+    kube_scheduler: "registry.fedoraproject.org/f27/kubernetes-scheduler"
+
+- name: Define kube 1.7 worker nodes and etcd images
+  when: ansible_distribution == "Fedora" and ansible_distribution_major_version > '26'
+  set_fact:
+    kube_maj_ver: "1.7"
+    kube_kubelet: "registry.fedoraproject.org/f27/kubernetes-kubelet"
+    kube_proxy: "registry.fedoraproject.org/f27/kubernetes-proxy"
+    etcd: "registry.fedoraproject.org/f27/etcd"
+
+
+- name: Install master node system containers that is, k8-apiserver, k8-controller_manager and k8-scheduler
+  when: ansible_distribution == "Fedora" and ansible_distribution_major_version > '26'
+  command: {{ item }}
+  with_items:
+    - atomic install --system --system-package=no --name kube-apiserver {{ kube_apiserver }}
+    - atomic install --system --system-package=no --name kube-controller-manager {{ kube_controller_manager }}
+    - atomic install --system --system-package=no --name kube-scheduler {{ kube_scheduler }}
+
+- name: Install worker node system containers, that is k8-kubelet, k8-proxy and etcd
+  when: ansible_distribution == "Fedora" and ansible_distribution_major_version > '26'
+  command: {{ item }}
+  with_items:
+    - atomic install --system --system-package=no --name kubelet {{ kube_kubelet }}
+    - atomic install --system --system-package=no --name kube-proxy {{ kube_proxy }}
+    - atomic install --system --system-package=no --storage=ostree --name etcd {{ etcd }}
+
```

Comment:
All these changes could be contained in a `block:` statement, so that we are only doing the `when:` check a single time.